include/m_field_info.h

```diff
@@ -7,10 +7,37 @@
 
 struct Game_Play;
 
+typedef enum fieldType {
+  /* 0 */ mFI_FIELDTYPE_FG,
+  /* 1 */ mFI_FIELDTYPE_1,
+  /* 2 */ mFI_FIELDTYPE_2,
+  /* 3 */ mFI_FIELDTYPE_ROOM,
+  /* 4 */ mFI_FIELDTYPE_NPC_ROOM,
+  /* 5 */ mFI_FIELDTYPE_DEMO,
+  /* 6 */ mFI_FIELDTYPE_PLAYER_ROOM,
+  /* 7 */ mFI_FIELDTYPE_NUM,
+}fieldType;
+
+
+#define mFI_TO_FIELD_ID(type, index) (((type) << 12) | (index))
```

Comment:
Macros should be UPPER_CASE
```suggestion
#define MFI_TO_FIELD_ID(type, index) (((type) << 12) | (index))
```